r/src/type_infer.cpp

```diff
@@ -72,7 +72,8 @@ std::shared_ptr<arrow::DataType> InferArrowTypeFromVector<INTSXP>(SEXP x) {
   } else if (Rf_inherits(x, "POSIXct")) {
     auto tzone_sexp = Rf_getAttrib(x, symbols::tzone);
     if (Rf_isNull(tzone_sexp)) {
```

Comment:
I think we need to do something here when `tzone_sexp = ""`. I think we are currently not catching `TZ = ""`

Reply:
It looks like we do not need to handle `tzone_sexp = ""` since both `TZ = NA` and `TZ = ""` result in the `"tzone"` attribute being NULL. Added a couple of expectations to reflect this.